images/capi/ansible/windows/roles/systemprep/tasks/main.yml

```
@@ -13,7 +13,23 @@
 # limitations under the License.
 ---
 # This file was adapted from https://github.com/Azure/aks-engine/blob/master/vhd/packer/configure-windows-vhd.ps1 for ansible
-- name: Disable automatic windows updates
+- name: Remove Windows updates default settings
+  win_regedit:
+    path: HKLM:\SOFTWARE\Policies\Microsoft\Windows\WindowsUpdate\
+    state: absent
+    delete_key: yes
+
+- name: Reset Windows updates path
+  win_regedit:
+    path: HKLM:\SOFTWARE\Policies\Microsoft\Windows\WindowsUpdate
+    state: present
+
+- name: Add Windows update path
+  win_regedit:
+    path: HKLM:\SOFTWARE\Policies\Microsoft\Windows\WindowsUpdate\AU
+    state: present
```

Can this just be 

```suggestion
- name: Add Windows update registry path
  win_regedit:
    path: HKLM:\SOFTWARE\Policies\Microsoft\Windows\WindowsUpdate\AU
    state: present
```

or does it not like creating the two locations at once?

---

also i think specifying registry path helps because cold reading without the info looks like its doing something to a folder path

---

> or does it not like creating the two locations at once?

It fails with permissions error if I don't set the `HKLM:\SOFTWARE\Policies\Microsoft\Windows\WindowsUpdate` path separately. I found this: https://github.com/MicrosoftDocs/windowsserverdocs/issues/2074#issuecomment-612141796